src/protocol/basics/sum_of_product/malicious.rs

```diff
@@ -60,8 +62,8 @@ impl AsRef<str> for Step {
 pub async fn sum_of_products<F>(
     ctx: MaliciousContext<'_, F>,
     record_id: RecordId,
-    a: &[&MaliciousReplicated<F>],
-    b: &[&MaliciousReplicated<F>],
+    a: &[MaliciousReplicated<F>],
+    b: &[MaliciousReplicated<F>],
```

Comment:
Looking at this, why can't `sum_of_products()` take `impl IntoIterator` so that we don't even need to `collect()` at the call site?